contracts/external/proxies/CERC20Proxy.sol

```diff
@@ -0,0 +1,90 @@
+
+import {ReentrancyGuard} from "../../packages/oz/ReentrancyGuard.sol";
+import {SafeERC20} from "../../packages/oz/SafeERC20.sol";
+import {ERC20Interface} from "../../interfaces/ERC20Interface.sol";
+import {Actions} from "../../libs/Actions.sol";
+import {Controller} from "../../Controller.sol";
+import {CERC20Interface} from "../../interfaces/CERC20Interface.sol";
+
+/**
+ * @title CERC20Proxy
+ * @author Opyn Team
+ * @dev Contract for wrapping cToken before minting options
+ */
+contract CERC20Proxy is ReentrancyGuard {
+    using SafeERC20 for ERC20Interface;
+    using SafeERC20 for CERC20Interface;
+
+    Controller public controller;
+    address public marginPool;
+
+    constructor(address _controller, address _marginPool) public {
+        controller = Controller(_controller);
+        marginPool = _marginPool;
+    }
+
+    /**
+     * @notice execute a number of actions after minting some cTokens
+     * @dev a wrapper for the Controller operate function, to wrap uderlying to cToken before the excution.
+     * @param _actions array of actions arguments
+     * @param _underlying underlying asset
+     * @param _cToken the cToken to mint
+     * @param _amountUnderlying the amount of underlying to supply to Compound
+     */
+    function operate(
+        Actions.ActionArgs[] memory _actions,
+        address _underlying,
+        address _cToken,
+        uint256 _amountUnderlying
+    ) external nonReentrant {
+        ERC20Interface underlying = ERC20Interface(_underlying);
+        CERC20Interface cToken = CERC20Interface(_cToken);
+
+        // if depositing token: pull token from user
+        uint256 cTokenBalance = 0;
+        if (_amountUnderlying > 0) {
+            underlying.safeTransferFrom(msg.sender, address(this), _amountUnderlying);
+            // mint cToken
+            underlying.safeIncreaseAllowance(address(_cToken), _amountUnderlying);
+
+            require(cToken.mint(_amountUnderlying) == 0, "CERC20Proxy: cToken mint failed");
+
+            cTokenBalance = cToken.balanceOf(address(this));
+            cToken.safeIncreaseAllowance(marginPool, cTokenBalance);
+        }
+
+        // verify sender
+        for (uint256 i = 0; i < _actions.length; i++) {
+            Actions.ActionArgs memory action = _actions[i];
+
+            // check that msg.sender is an owner or operator
+            if (action.owner != address(0)) {
+                require(
+                    (msg.sender == action.owner) || (controller.isOperator(action.owner, msg.sender)),
+                    "CERC20Proxy: msg.sender is not owner or operator "
+                );
+            }
+
+            // overwrite the deposit amount by the exact amount minted
+            if (action.actionType == Actions.ActionType.DepositCollateral && action.amount == 0) {
+                // should we only update if the action.second address == address(this)?
+                _actions[i].amount = cTokenBalance;
+            }
```

Comment:
```suggestion
            if (action.actionType == Actions.ActionType.DepositCollateral && action.amount == 0) {
                _actions[i].amount = cTokenBalance;
            }
```
Don't need to add that condition, if user has cToken just not use action.amount = 0

Reply:
Can remove the comment

Reply:
Yeah that was internal monologue that I forgot to delete :)